tests/graph.rs

```diff
@@ -342,6 +343,62 @@ fn dijk() {
     assert_eq!(scores[&c], 9);
 }
 
+#[test]
+fn test_astar_null_heuristic() {
+    let mut g = Graph::new();
+    let a = g.add_node("A");
+    let b = g.add_node("B");
+    let c = g.add_node("C");
+    let d = g.add_node("D");
+    let e = g.add_node("E");
+    let f = g.add_node("F");
+    g.add_edge(a, b, 7);
+    g.add_edge(c, a, 9);
+    g.add_edge(a, d, 14);
+    g.add_edge(b, c, 10);
+    g.add_edge(d, c, 2);
+    g.add_edge(d, e, 9);
+    g.add_edge(b, f, 15);
+    g.add_edge(c, f, 11);
+    g.add_edge(e, f, 6);
+
+    let path = astar(&g, a, |finish| finish == e, |e| *e.weight(), |_| 0);
+    assert_eq!(path, Some((23, vec![a, d, e])));
+
+    let path = astar(&g, e, |finish| finish == b, |e| *e.weight(), |_| 0);
+    assert_eq!(path, None);
+}
+
+#[test]
+fn test_astar_manhattan_heuristic() {
+    let mut g = Graph::new();
+    let a = g.add_node((0., 0.));
+    let b = g.add_node((2., 0.));
+    let c = g.add_node((1., 1.));
+    let d = g.add_node((0., 2.));
+    let e = g.add_node((3., 3.));
+    let f = g.add_node((4., 2.));
+    g.add_edge(a, b, 2.);
+    g.add_edge(a, d, 4.);
+    g.add_edge(b, c, 1.);
+    g.add_edge(b, f, 7.);
+    g.add_edge(c, e, 5.);
+    g.add_edge(e, f, 1.);
+    g.add_edge(d, e, 1.);
+
+    let path = astar(&g, a, |finish| finish == f, |e| *e.weight(),
+    |node| {
+        let (x1, y1): (f32, f32) = g[node];
+        let (x2, y2): (f32, f32) = g[f];
+
+        let (x, y) = (x2 + x1, y2 + y1);
+
+        x.powi(2) + y.powi(2)
```

Comment:
Euclidean distance is sqrt((x2 - x1)² + (y2 - y1)²) and is that what this is going for?